### PR TITLE
https://issues.redhat.com/browse/ACM-9984 osImages auth docs

### PR DESCRIPTION
**Pull request by oafischer**
2.10 only

### CODE REVIEW

File: clusters/cluster_lifecycle/cim_enable.adoc [hunk 1]
```diff
@@ -127,6 +127,32 @@ data:
 +
 Registries in the list of `unqualified-search-registries` are automatically added to an authentication ignore list in the `PUBLIC_CONTAINER_REGISTRIES` environment variable. The specified registries do not require authentication when the pull secret of the managed cluster is validated.
 
+. Write the key pairs representing the headers and query parameters that you want to send with every `osImage` request. If you don't need both parameters, you can write key pairs for only headers or query parameters.
+
+.. Create a file named `headers` and add content that resembles the following example:
```
Comment (paul-maidment) [inline clusters/cluster_lifecycle/cim_enable.adoc:134]: Maybe add a line to make it clear that `headers` and `query_params` are the only names accepted for these files.

File: clusters/release_notes/whats_new.adoc [hunk 2]
```diff
@@ -22,6 +22,8 @@ Learn about what's new relating to Cluster lifecycle with {mce-short}.
 
 - *Technology Preview:* You can now customize the server URL and hub cluster API CA bundle so that you can import a managed cluster on your {mce-short} hub cluster, even if you have intermediate components. See xref:../cluster_lifecycle/adv_config_cluster.adoc#custom-server-url-ca[Customizing the server URL and hub cluster API CA bundle (Technology Preview)] to learn more.
 
+- The `osImages` spec now supports authentication. See xref:../cluster_lifecycle/cim_enable.adoc#enable-cim-disconnected[Enabling central infrastructure management in disconnected environments] to learn more.
```
Comment (paul-maidment) [inline clusters/release_notes/whats_new.adoc]: Strictly speaking we support the passing of HTTP Headers and Query Parameters in each request to fetch an OS image. This was with the intent of allowing users to make use of simple HTTP authentication schemes.

You should also add a note that we only recommend using this feature on HTTPS endpoints as HTTP (without the S) is unencrypted on the wire.

File: clusters/cluster_lifecycle/cim_enable.adoc [hunk 3]
```diff
@@ -127,6 +127,34 @@ data:
 +
 Registries in the list of `unqualified-search-registries` are automatically added to an authentication ignore list in the `PUBLIC_CONTAINER_REGISTRIES` environment variable. The specified registries do not require authentication when the pull secret of the managed cluster is validated.
 
+. Write the key pairs representing the headers and query parameters that you want to send with every `osImage` request. If you don't need both parameters, you can write key pairs for only headers or query parameters.
+
+*Important:* Headers and query parameters are only encyrpted if you use HTTPS. Make sure to use HTTPS to avoid security issues.
```
Comment (paul-maidment) [inline clusters/cluster_lifecycle/cim_enable.adoc]: `encyrpted`
should be `encrypted`

**Review by paul-maidment — APPROVED** [review]
/lgtm

File: clusters/cluster_lifecycle/cim_enable.adoc [hunk 4]
```diff
@@ -127,6 +127,34 @@ data:
 +
 Registries in the list of `unqualified-search-registries` are automatically added to an authentication ignore list in the `PUBLIC_CONTAINER_REGISTRIES` environment variable. The specified registries do not require authentication when the pull secret of the managed cluster is validated.
 
+. Write the key pairs representing the headers and query parameters that you want to send with every `osImage` request. If you don't need both parameters, you can write key pairs for only headers or query parameters.
```
Comment (jc-berger) [inline clusters/cluster_lifecycle/cim_enable.adoc]: Minor change for minimalism: 

```suggestion
. Write the key pairs representing the headers and query parameters that you want to send with every `osImage` request. If you don't need both parameters, write key pairs for only headers or query parameters.
```

File: clusters/release_notes/whats_new.adoc [hunk 5]
```diff
@@ -22,6 +22,8 @@ Learn about what's new relating to Cluster lifecycle with {mce-short}.
 
 - *Technology Preview:* You can now customize the server URL and hub cluster API CA bundle so that you can import a managed cluster on your {mce-short} hub cluster, even if you have intermediate components. See xref:../cluster_lifecycle/adv_config_cluster.adoc#custom-server-url-ca[Customizing the server URL and hub cluster API CA bundle (Technology Preview)] to learn more.
 
+- You can now pass HTTP/HTTPS headers and query parameters in each request to fetch an OS image. See xref:../cluster_lifecycle/cim_enable.adoc#enable-cim-disconnected[Enabling central infrastructure management in disconnected environments] to learn more.
```
Comment (jc-berger) [inline clusters/release_notes/whats_new.adoc]:
```suggestion
- You can now pass HTTP/HTTPS headers and query parameters in each request to fetch an OS image. To learn more, see xref:../cluster_lifecycle/cim_enable.adoc#enable-cim-disconnected[Enabling central infrastructure management in disconnected environments].
```

**Review by jc-berger — APPROVED** [review]
/lgtm

**Pull request merged**